src/libs/TransactionRecoveryApi/AxelarRecoveryApi.ts

```diff
@@ -71,34 +72,37 @@ export class AxelarRecoveryApi {
   }
 
   public async queryTransactionStatus(txHash: string): Promise<GMPStatusResponse> {
+    let isError = false;
+    let status = GMPStatus.ERROR_FETCHING_STATUS;
 
     const txDetails = await this.execGet(this.axelarCachingServiceUrl, {
       method: "searchGMP",
       txHash,
-    }).then(res => res[0]);
+    })
+      .then((res) => res[0])
+      .catch((e) => {
+        isError = true;
+      });
 
-    if (!txDetails) return { status: "error_fetching_status" }
+    if (!txDetails || isError) return { status };
 
     const { approved, call, gas_status, gas_paid, is_executed, error } = txDetails;
 
     const gasPaidInfo: GasPaidInfo = {
       status: gas_status,
-      details: gas_paid
+      details: gas_paid,
     };
 
-    const gmpStatusDetails: GMPStatusDetails = {
-      srcGatewayCalled: !!call,
-      destGatewayApproved: !!approved,
-      destExecuted: is_executed,
-      destAppError: error,
-      gasPaidInfo
-    }
+    if (is_executed) status = GMPStatus.DEST_EXECUTED
+    else if (approved) status = GMPStatus.DEST_GATEWAY_APPROVED
+    else if (call) status = GMPStatus.SRC_GATEWAY_CALLED
 
     return {
-      status: gmpStatusDetails,
-      call_data: txDetails[0]?.call
-    }
-
+      status,
+      errors: error,
```

Comment:
`errors` -> `error` i think we will have a single error

Reply:
this is an array of errors that i just pass on directly from the downstream API call.